packages/graph/hash_graph/lib/graph/src/knowledge/entity/query.rs

```diff
@@ -63,7 +66,155 @@ impl RecordPath for EntityQueryPath<'_> {
+        self.position += 1;
+        Ok(match token {
+            EntityQueryToken::Id => EntityQueryPath::Id,
+            EntityQueryToken::OwnedById => EntityQueryPath::OwnedById,
+            EntityQueryToken::CreatedById => EntityQueryPath::CreatedById,
+            EntityQueryToken::UpdatedById => EntityQueryPath::UpdatedById,
+            EntityQueryToken::RemovedById => EntityQueryPath::RemovedById,
+            EntityQueryToken::Version => EntityQueryPath::Version,
+            EntityQueryToken::Type => {
+                let entity_type_query_path =
+                    EntityTypeQueryPathVisitor::new(self.position).visit_seq(seq)?;
+
+                EntityQueryPath::Type(entity_type_query_path)
+            }
+            EntityQueryToken::Properties => {
+                let property = seq
+                    .next_element()?
+                    .ok_or_else(|| de::Error::invalid_length(self.position, &self))?;
+                EntityQueryPath::Properties(Some(property))
+            }
+            EntityQueryToken::OutgoingLinks | EntityQueryToken::IncomingLinks => {
+                todo!("https://app.asana.com/0/0/1203167266370359/f")
+            }
+        })
+    }
+}
+impl<'de: 'k, 'k> Deserialize<'de> for EntityQueryPath<'k> {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        deserializer.deserialize_seq(EntityQueryPathVisitor::new(0))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::query::test_utils::create_path;
+
+    fn convert_path(segments: impl IntoIterator<Item = &'static str>) -> EntityQueryPath<'static> {
+        EntityQueryPath::try_from(create_path(segments)).expect("Could not convert path")
+    }
+
+    fn deserialize<'q>(segments: impl IntoIterator<Item = &'q str>) -> EntityQueryPath<'q> {
+        EntityQueryPath::deserialize(de::value::SeqDeserializer::<_, de::value::Error>::new(
+            segments.into_iter(),
+        ))
+        .expect("Could not deserialize path")
+    }
+
+    #[test]
+    fn deserialization() {
+        assert_eq!(deserialize(["version"]), EntityQueryPath::Version);
+        assert_eq!(deserialize(["ownedById"]), EntityQueryPath::OwnedById);
+        assert_eq!(
+            deserialize(["type", "version"]),
+            EntityQueryPath::Type(EntityTypeQueryPath::Version)
+        );
+        assert_eq!(
+            deserialize([
+                "properties",
+                "https://blockprotocol.org/@alice/types/property-type/name/"
+            ]),
+            EntityQueryPath::Properties(Some(Cow::Borrowed(
+                "https://blockprotocol.org/@alice/types/property-type/name/"
+            )))
+        );
+
+        assert_eq!(
+            EntityQueryPath::deserialize(de::value::SeqDeserializer::<_, de::value::Error>::new(
+                ["version", "test"].into_iter()
+            ))
+            .expect_err("Could convert entity query path with multiple tokens")
```

Comment:
Can you update this one to be more verbose like the one we fixed quite a few PRs ago. Something like "could convert X even though we expected to not be able to because of Y" or whatever it was.